src/Native/System.Native/pal_networking.cpp

```diff
@@ -2376,7 +2376,7 @@ static void ConvertEventEPollToSocketAsync(SocketEvent* sae, epoll_event* epoll)
     uint32_t events = epoll->events;
     if ((events & EPOLLHUP) != 0)
     {
-        events = (events & ~EPOLLHUP) | EPOLLIN | EPOLLOUT;
+        events = (events & UnsignedCast(~EPOLLHUP)) | EPOLLIN | EPOLLOUT;
```

Comment:
I'm surprised this isn't triggering an assert. EPOLLHUP is 16 (at least on my Ubuntu install), which makes ~EPOLLHUP -17, and UnsignedCast asserts that the input is not negative.